src/models/base/FirestoreCollection.ts

```diff
@@ -14,8 +15,8 @@ export default abstract class FirestoreCollection<T> {
       .withConverter(this.converter);
   }
 
-  new(data?: T): Model<T> {
-    return new FirestoreModel<T>(this.collection.doc(), data ?? ({} as T));
+  new(data?: T): ScholarshipModel<T> {
+    return new ScholarshipModel<T>(this.collection.doc(), data ?? ({} as T));
```

Comment:
This is incorrect. **FirestoreCollection** is a generic class for _any_ kind of collection, not just the Scholarships one. You can override this method in the _Scholarship-specific_ defined subclass in `models/Scholarships.ts` instead.